File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/inputForm/InputContract.java
package com.non_name_hero.calenderview.inputForm;

import com.non_name_hero.calenderview.utils.BasePresenter;
import com.non_name_hero.calenderview.utils.BaseView;

import java.util.Calendar;

public interface InputContract {
    //viewへのの出力、入力のインターフェース
    interface View extends BaseView<InputContract.Presenter> {
        void finishInput();
    }

    //Presenterへの入力、出力のインターフェース
    interface Presenter extends BasePresenter {
        void saveSchedule(String title, String description, Calendar startAtSchedule, Calendar endAtSchedule, int groupId, int paymentId);
    }

}

File: CalenderView/app/src/main/java/com/non_name_hero/calenderview/inputForm/InputContract.kt
package com.non_name_hero.calenderview.inputForm

import com.non_name_hero.calenderview.utils.BasePresenter
import com.non_name_hero.calenderview.utils.BaseView
import java.util.*

interface InputContract {
    //viewへのの出力、入力のインターフェース
    interface View : BaseView<Presenter?> {
        fun finishInput()
    }

    //Presenterへの入力、出力のインターフェース
    interface Presenter : BasePresenter {
        fun saveSchedule(title: String?, description: String?, startAtSchedule: Date?, endAtSchedule: Date?, groupId: Int, paymentId: Int)
        fun saveBalance(price: Long, balanceCategoryId: Int, usedAtDatetime: Date, title: String?)
    }
}